src/resolve-config.js

```diff
@@ -1,31 +1,76 @@
 "use strict";
 
 const cosmiconfig = require("cosmiconfig");
+const editorconfig = require("editorconfig");
+const mem = require("mem");
 const minimatch = require("minimatch");
 
 const asyncWithCache = cosmiconfig("prettier");
 const asyncNoCache = cosmiconfig("prettier", { cache: false });
 const syncWithCache = cosmiconfig("prettier", { sync: true });
 const syncNoCache = cosmiconfig("prettier", { cache: false, sync: true });
 
+const editorconfigAsyncNoCache = filePath =>
+  filePath && editorconfig.parse(filePath).then(editorConfigToPrettier);
+const editorconfigAsyncWithCache = mem(editorconfigAsyncNoCache);
+const editorconfigSyncNoCache = filePath =>
+  filePath && editorConfigToPrettier(editorconfig.parseSync(filePath));
+const editorconfigSyncWithCache = mem(editorconfigSyncNoCache);
+
+function editorConfigToPrettier(editorConfig) {
+  const result = {};
+
+  if (editorConfig.indent_style) {
+    result.useTabs = editorConfig.indent_style === "tab";
+  }
+
+  const tabWidth = editorConfig.indent_size || editorConfig.tab_width;
+  if (tabWidth) {
+    result.tabWidth = tabWidth;
+  }
+
+  if (editorConfig.max_line_length) {
+    result.printWidth = editorConfig.max_line_length;
+  }
+
+  return result;
+}
+
 function resolveConfig(filePath, opts) {
   const useCache = !(opts && opts.useCache === false);
-  return (useCache ? asyncWithCache : asyncNoCache)
-    .load(filePath)
-    .then(result => {
-      return !result ? null : mergeOverrides(result.config, filePath);
-    });
+  return Promise.all([
+    (useCache ? asyncWithCache : asyncNoCache).load(filePath),
+    (useCache ? editorconfigAsyncWithCache : editorconfigAsyncNoCache)(filePath)
+  ]).then((arr /* [result, editorConfigged] */) =>
+    mergeEditorConfig(arr[0], filePath, arr[1])
```

Comment:
Can we put `filePath` arg in front of `arr[x]`? as it looks more comfortable and reasonable (`result` and `editorConfigged` are config-related).

Reply:
Sure, fixed in https://github.com/prettier/prettier/pull/2760/commits/5c9c3d9fea6795ef7a314164ff05f28b62f15994. I also made the variable names explicit in https://github.com/prettier/prettier/pull/2760/commits/f69d367ef3a943c3c704a35076adfedbcde52999